# Hypothesis Testing Assignment 3




1. A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.

Minitab File : Cutlets.mtw

Ans:
Let, H0:there is no significant difference in the diameter of the cutlet between two units
                                          vs
     H1:there is any significant difference in the diameter of the cutlet between two units
 Here,Two sample two tail test is apllicable.
  
 alpha=0.05 

In [11]:
import pandas as pd
from scipy import stats
Data=pd.read_csv('/content/Cutlets.csv')
Data.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [12]:
p_value=stats.ttest_ind(Data['Unit A'],Data['Unit B']).pvalue
p_value

0.4722394724599501

    p-value = 0.4722394
    p-value = 0.4722394 > alpha = 0.05 since, we do not reject the null hypothesis.
    Therefore, there is no significant difference in the diameter of the cutlet between two units.

2. A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch.
    Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.
 
    Minitab File: LabTAT.mtw

In [13]:
import pandas as pd
from scipy import stats
lab_data=pd.read_csv('/content/LabTAT.csv')
lab_data.head()


Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [14]:
stats.f_oneway(lab_data['Laboratory 1'],lab_data['Laboratory 2'],lab_data['Laboratory 3'],lab_data['Laboratory 4']).pvalue

2.1156708949992414e-57

p-value = 2.1156708949992414e-57

p-value < 0.05,

So we reject null hypothesis at 5% l.o.s. Since,there is difference in the average TAT among the different laboratories.



3. Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.
              East	  West	North	South
        Males       50    142 	131 	70
        Females  	550	  351	480 	350


In [15]:
import pandas as pd
from scipy import stats
buyer_ratio =pd.read_csv('/content/BuyerRatio.csv')
buyer_ratio

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [16]:
buyer_ratio=buyer_ratio.iloc[:,1:]
buyer_ratio

East  West  North  South
0    50   142    131     70
1   435  1523   1356    750

In [17]:
stats.chi2_contingency(buyer_ratio)


Chi2ContingencyResult(statistic=1.595945538661058, pvalue=0.6603094907091882, dof=3, expected_freq=array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
       [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

p-value =0.6603094907091882

p-value > 0.05,

We do not reject null hypothesis and conlude that, There is no significant difference between genders and regions.

4. TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences.

    Minitab File: CustomerOrderForm.mtw


In [18]:
import pandas as pd
import numpy as np
import scipy as sp
from scipy import stats


In [19]:
Cust =pd.read_csv('/content/Costomer+OrderForm.csv')
Cust.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [20]:
Phillippines=Cust['Phillippines'].value_counts()
Phillippines

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [21]:
Indonesia=Cust['Indonesia'].value_counts()
Indonesia

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [22]:
Malta=Cust['Malta'].value_counts()
Malta

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [23]:
India=Cust['India'].value_counts()
India

Error Free    280
Defective      20
Name: India, dtype: int64

In [24]:
stats.chi2_contingency([[271,267,269,280],[29,33,31,20]])

Chi2ContingencyResult(statistic=3.858960685820355, pvalue=0.2771020991233135, dof=3, expected_freq=array([[271.75, 271.75, 271.75, 271.75],
       [ 28.25,  28.25,  28.25,  28.25]]))

 p-value = 0.2771020991233135
 
 p-value > 0.05,

 We do not reject null hypothesis and conlude that, There is no significant difference between Error Free and Defectives.